packages/wallet-sdk/src/core/communicator/Communicator.test.ts

```diff
@@ -72,10 +75,26 @@ describe('Communicator', () => {
       },
       urlOrigin
     );
+    expect(popup).toBeTruthy();
+  });
+
+  test('@onMessage - should add and remove event listener', async () => {
+    const mockRequest: Message = {
+      requestId: 'mock-request-id-1-2',
+      data: 'test',
+    };
+
+    queueMessageEvent({ data: mockRequest });
+
+    const promise = communicator.onMessage(() => true);
+
+    expect(addEventListenerSpy).toHaveBeenCalled();
+    expect(await promise).toEqual(mockRequest);
+    expect(removeEventListenerSpy).toHaveBeenCalled();
   });
 
-  it('should post a message to the popup window', async () => {
-    const mockRequest: Message = { id: 'mock-request-id-1-2', data: {} };
+  test('@postRequestAndWaitForResponse - should post a message to the popup window and wait for response', async () => {
```

Comment:
instead of @, coudl these be `describe('postRequestAndWaitForResponse', () => {});` for consistency?